auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -73,6 +76,42 @@ public Property(String humanName, ExecutableElement element) {
 
       type = TypeName.get(element.getReturnType());
       annotations = buildAnnotations(element);
+
+      TypeMirror typeAdapterAnnotation = getAnnotationValue(element, GsonTypeAdapter.class);
+      if (typeAdapterAnnotation == null) {
+        typeAdapter = null;
+      } else {
+        typeAdapter = TypeName.get(typeAdapterAnnotation);
+      }
+    }
+
+    public static TypeMirror getAnnotationValue(Element foo, Class<?> annotation) {
```

Comment:
It seems like this is the only way of getting the annotation value without catching exceptions, but perhaps there is a better way.